File: src/main/csharp/p00/Problem0010.cs
using System;

namespace ProjectEuler.main.csharp.p00
{
    public class Problem0010
    {
        public static long Solve()
        {
            var primeSieve = new bool[2000000];
            var limit = Math.Sqrt(primeSieve.Length);
            long sum = 2;

            for (var i = 3; i < limit; i += 2)
                if (!primeSieve[i])
                {
                    sum += i;

                    for (var j = i * i; j < 2000000; j += i * 2) primeSieve[j] = true;
                }

            for (var i = (int) limit + 1; i < 2000000; i += 2)
                if (!primeSieve[i])
                    sum += i;


            return sum;
        }
    }
}

File: src/main/kotlin/p00/Problem0010.kt
package p00

import kotlin.math.sqrt

class Problem0010 {

    fun solve(): Long {
        val toN = 2000000
        val values = BooleanArray(toN)
        val sqrt = sqrt(toN.toDouble()).toInt()
        var sum = 2L

        for (i in 3..sqrt step 2)
            if (!values[i]) {
                sum += i
                for (j in i * i until toN step i + i) values[j] = true
            }
        for (i in sqrt + 1 until toN step 2) if (!values[i]) sum += i

        return sum
    }

}